proposals/0278-screenmanager-layout-management.md

```diff
@@ -11,33 +11,78 @@ Adds a screen manager method for changing the current layout to a new layout.
 The manager layer should be able to handle most tasks that used to be handled by RPCs, and the Screen Manager in particular should be able to handle manipulating the screen and templates without requiring the developer to fall back to RPCs in any cases.
 
-One big area that it is currently missing is the ability to manage changing templates, currently done with the `SetDisplayLayout` RPC (on RPC spec <6.0) and `Show` RPC (on RPC spec >6.0). This proposal adds a way to change the layout while taking care of backward compatibility concerns.
+One big area that it is currently missing is the ability to manage changing templates, currently done with the `SetDisplayLayout` RPC (on RPC spec <6.0) and `Show` RPC (on RPC spec >=6.0). This proposal adds a way to change the layout while taking care of backward compatibility concerns.
 
 ## Proposed solution
-We will add a new screen manager method for changing the current layout to a new layout.
+We will add a new screen manager method for changing the current layout to a new layout. This also includes updating the template's night and day color schemes when the feature is available on systems supporting RPC spec 5.0.0 or newer. 
 
 ### iOS APIs
 ```objc
 @interface SDLScreenManager : NSObject
 
-/// Change the current layout to a new layout. You can get a list of SDL-defined layout strings in SDLPredefinedLayout.
-- (void)changeLayout:(NSString *)newLayout withCompletionHandler:(SDLScreenManagerUpdateCompletionHandler)handler;
+/// Change the current layout to a new layout, update the layout's night and day color schemes and update the text, graphics, buttons and template title.
+- (void)changeLayout:(SDLPredefinedLayout)newLayout templateUpdates:(nullable SDLTemplateUpdates *)templateUpdates withUpdateCompletionHandler:(SDLScreenManagerUpdateCompletionHandler)handler;
+
+@end
+
+@interface SDLTemplateUpdates : NSObject
+
+@property (copy, nonatomic, nullable) SDLTemplateColorScheme *dayColorScheme;
+@property (copy, nonatomic, nullable) SDLTemplateColorScheme *nightColorScheme;
+@property (copy, nonatomic, nullable) NSString *textField1;
+@property (copy, nonatomic, nullable) NSString *textField2;
+@property (copy, nonatomic, nullable) NSString *textField3;
+@property (copy, nonatomic, nullable) NSString *textField4;
+@property (copy, nonatomic, nullable) NSString *mediaTrackTextField;
+@property (strong, nonatomic, nullable) SDLArtwork *primaryGraphic;
+@property (strong, nonatomic, nullable) SDLArtwork *secondaryGraphic;
+@property (copy, nonatomic) SDLTextAlignment textAlignment;
+@property (copy, nonatomic, nullable) SDLMetadataType textField1Type;
+@property (copy, nonatomic, nullable) SDLMetadataType textField2Type;
+@property (copy, nonatomic, nullable) SDLMetadataType textField3Type;
+@property (copy, nonatomic, nullable) SDLMetadataType textField4Type;
+@property (copy, nonatomic, nullable) NSString *title;
+@property (copy, nonatomic) NSArray<SDLSoftButtonObject *> *softButtonObjects;
 
```

Comment:
Add initializers for creating including the required layout name with either a string or predefined layout. It's probably fine to only have one initializer with the custom string and make someone who uses it set up the rest of it separately by setting properties. That way you don't have to duplicate every initializer. I would have something like:

1. Initializer with just String layout
2. Initializer with just PredefinedLayout
3. Initializer with PredefinedLayout, text fields 1-4, title, softButtonObjects
4. One with everything